### DISCUSSION TITLE
v1.0.0 Limitations Text

### DISCUSSION THREAD

**Discussion by GeoffLyle** [CBIIT/ccdi-federation-api#104] [API Design Discussion]
Current Draft of the Limitations of v1.0.0

## CCDI Federation open API v1.0.0 Limitations

CCDI Federation has done our best to ensure the v1.0.0 release is fully functional and provides useful and accessible information to users.  There are some limitations to this release that we would like to make known to users.

### Data Harmonization:
Where possible, data has been harmonized to Common Data Elements provided by the Cancer Data Standards Registry and Repository.  One exception is the `diagnosis` values are from the [CCDI Submission Template v1.7.2](https://github.com/CBIIT/ccdi-federation-api/blob/main/assets/CCDI_Submission_Template_v1.7.2.diagnosis_values.xlsx), which includes values from CNS Diagnosis Category, ICD-0 3.2, and WHO CNS5 Integrated Diagnosis.  Harmonized `diagnosis` values provided by Federation members are not always an exact match, and we recommend checking the unharmonized diagnosis values provided or contacting the member providing the data for further information.  Also, for the age data provided, some of the data has been converted from age in years to age in days, so this data should not be taken as exact values.

In future updates we plan to add more harmonized data fields regarding the anatomic site the sample was taken from, the library selection method used to enrich the molecules for high throughput sequencing, the cellular material used for sequencing, and the molecules extracted from that material.  

### Technical Aspects:
While the CCDI tests the API implementations of the Federation members, errors may occur in member servers that are beyond our control.  If you believe such an error has occurred when accessing a Federation member’s data, please contact the appropriate individual here: [LINK TO CONTACTS FOR CCDI MEMBERS]

The API interface is also currently limited in the filtering options available; filtering by AND is accepted, but not OR or other operands.  The API will also return results in an alphabetical order based on subject or sample ID with no other options to sort the data.  Summary statistics are only provided by the /count endpoint.

**Comment by mcrusch** [comment]
My perspective probably approximates a users', as I haven't been deeply involved along the way.  I find the content clear and useful.  I like the length.  I can't speak to the completeness.

A few minor suggestions:
* Break the age sentence into its own short paragraph.
* Move the filtering paragraph to a new 2nd section labeled "Filtering:"
* Relabel the last section as "Technical contacts" or "Troubleshooting" or something like that?  The idea that at any point in time, each implementation might have a bug or go down is not really a limitation, per se.  After all, there can never be a version of the API that makes that go away.  But it is useful to point out that the implementations are tested and to provide the contacts so that people can know how to find help.  Hence, I think the content is good, and the section title can be tweaked.

**Comment by lydiabrogers** [comment]
Under Technical aspects can we add something along the lines that says: Compliance with the API specification and the data provided are the responsibility of the individual Federation members. i think it may be a good disclaimer as the first sentence.

**Comment by jsunny23** [comment]
> The API interface is also currently limited in the filtering options available; [...]

My only suggestion is to elaborate a bit on the filtering limitations to something like:

"The current filtering interface is through HTTP GET parameters, which are limited in scale and complexity; [...]"

And we could say something about updating these to POST parameters in future iterations to support large payloads and more intricate filtering requests.

**Comment by lydiabrogers** [comment]
i made change to above and attached in word doc.  Once final I want to send to Subhashini to get feedback.

[CCDI Federation open API v1 limitations.docx](https://github.com/CBIIT/ccdi-federation-api/files/15472005/CCDI.Federation.open.API.v1.limitations.docx)
